diff --git a/faster-rnnlm/rnnlm.cc b/faster-rnnlm/rnnlm.cc
index bf2bd7d..4675f71 100644
--- a/faster-rnnlm/rnnlm.cc
+++ b/faster-rnnlm/rnnlm.cc
@@ -445,9 +445,9 @@ void TrainLM(
     }
 
     Real ratio = bl_entropy / entropy;
-    if (isnan(entropy) || isinf(entropy) || !(ratio >= bad_ratio)) {
+    if (std::isnan(entropy) || std::isinf(entropy) || !(ratio >= bad_ratio)) {
       // !(ratio >= bad_ratio) will catch nan and inf even in fastmath mode
-      if (isnan(entropy) || isinf(entropy) || !(ratio >= awful_ratio)) {
+      if (std::isnan(entropy) || std::isinf(entropy) || !(ratio >= awful_ratio)) {
         fprintf(stderr, "\tAwful: Nnet rejected");
         nnet->ReLoad(model_weight_file);
       } else {
